test_Budget_pytest.py

```diff
@@ -1,96 +1,167 @@
-import pytest
-import click
-import csv
-from click.testing import CliRunner
-from main import clack
+""" Pytest document testing a few basic functionalities"""
+
+import json
+import logging
+import os
+import tempfile
+import time
+from typing import Literal
+from unittest.mock import patch
+
 from main import (
     Expense,
-    read_db_or_init,
-    find_next_id,
-    export_python,
+    add,
     add_csv_to_db,
-    report,
-    print_expenses,
-    import_csv,
     add_expense,
-    add,
+    clack,
+    export_python,
+    find_next_id,
+    import_csv,
+    print_expenses,
+    read_db_or_init,
+    read_expenses,
+    report,
     save_db,
-    DB_FILENAME,
+    DB_FILENAME
 )
-import pickle
-from unittest.mock import mock_open, patch
-import sys
-import tempfile
-import os
-from io import StringIO
-
-
-@pytest.fixture
-def mock_expense_list():
-    return
 
+from click.testing import CliRunner
+import pytest
 
-def test_export_python():
-    runner = CliRunner()
-    mock_output = [
+MOCK_OUTPUT_2_ITEMS = [
         Expense(id=1, amount=34.0, description="Hit Box"),
         Expense(id=2, amount=453.0, description="Crazy curry"),
     ]
-    with patch("main.read_db_or_init", return_value=mock_output):
+
+LIST_OF_ITEMS = [(987,"Chicken"),(456,"Bananas"),(789,"Pizza")]
+
+
+@pytest.fixture
+def runner():
+   return CliRunner() 
+
+@pytest.fixture
+def use_temp_csv():
+    filename_csv = "temp.csv"
+    with open(filename_csv, "w") as file:
+        file.write('amount,description\n150,Test CSV 1\n300,Test CSV 2\n')  
+    yield filename_csv
+    os.remove(filename_csv)
+
+
+@pytest.fixture
+def use_temp_db():
+    filename_db = "test_db.json"
+    with open(filename_db, "w") as file:
+        file.write("[]")  
+    yield filename_db
+    os.remove(filename_db)
+
+# @pytest.fixture
```

Comment:
To można odkomentować i dopisać podobną funkcję dla pliku csv :). Następnie wrzucić to jako argument do funkcji testowych.